File: bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      silence?(input) -> "Fine. Be that way!"
      shooting_asking?(input) -> "Calm down, I know what I'm doing!"
      shooting?(input) -> "Whoa, chill out!"
      asking?(input) -> "Sure."
      true -> "Whatever."
    end
  end

  defp shooting?(input) do
    String.upcase(input) == input && String.downcase(input) != input
  end
  defp asking?(input) do
    String.last(input)  == "?"
  end
  defp silence?(input) do
    String.trim(input) == ""
  end
  defp shooting_asking?(input) do
    shooting?(input) and asking?(input)
  end
end

File: bob/src/bob.erl
-module(bob).

-export([response/1]).


yelling("") -> false;
yelling(S) ->
    (string:to_upper(S) == S) and not (string:to_lower(S) == S).

question("") -> false;
question(S) ->
    lists:last(S) == $?.

response(String) ->
    Trimed = string:trim(String),
    Yelling = yelling(Trimed),
    Question = question(Trimed),
    if
        Question andalso Yelling-> "Calm down, I know what I'm doing!";
        Question -> "Sure.";
        Yelling -> "Whoa, chill out!";
        Trimed == "" -> "Fine. Be that way!";
        true -> "Whatever."
    end
        .
